src/main/java/mahaveer/Storage.java

```diff
@@ -0,0 +1,226 @@
+        default:
+            return null;
+        }
+        task.setDone(isDone);
+        return task;
+    }
+
+    /**
+     * Saves the provided list of tasks to the storage file.
+     *
+     * @param taskList the list of tasks to be saved.
+     */
+    private void saveTasks(List<Task> taskList) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH))) {
+            for (Task task : taskList) {
+                writer.write(task.toFileFormat());
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving tasks: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Adds a new todo task to the storage.
+     *
+     * @param description the description of the todo task.
+     */
+    public void addTodo(String description) {
+        List<Task> taskList = loadTasks();
+        taskList.add(new Task(description));
+        saveTasks(taskList);
+    }
+
+    /**
+     * Adds a new deadline task to the storage.
+     *
+     * @param description the task description.
+     * @param by          the deadline time.
+     */
+    public void addDeadline(String description, String by) {
+        List<Task> taskList = loadTasks();
+        taskList.add(new Deadline(description, by));
+        saveTasks(taskList);
+    }
+
+    /**
+     * Adds a new event task to the storage.
+     *
+     * @param description the task description.
+     * @param from        the event start time.
+     * @param to          the event end time.
+     */
+    public void addEvent(String description, String from, String to) {
+        List<Task> taskList = loadTasks();
+        taskList.add(new Event(description, from, to));
+        saveTasks(taskList);
+    }
+
+    /**
+     * Deletes a task at the specified index from the storage.
+     *
+     * @param index the index of the task to be deleted.
+     */
+    public void deleteTask(int index) {
+        List<Task> taskList = loadTasks();
+        if (index >= 0 && index < taskList.size()) {
+            taskList.remove(index);
+            saveTasks(taskList);
+        } else {
+            System.out.println("Task index out of bounds.");
+        }
+    }
+
+    /**
+     * Marks the task at the specified index as done in the storage.
+     *
+     * @param index the index of the task to mark as done.
+     */
+    public void markTask(int index) {
+        try {
+            List<String> lines = new ArrayList<>();
+            int currentIndex = 0;
+            try (BufferedReader reader = new BufferedReader(new FileReader(FILE_PATH))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    String[] parts = line.split(", ");
+                    if (currentIndex == index && parts.length >= 2) {
+                        parts[1] = "1"; // Update second column to "1"
+                        line = String.join(", ", parts);
+                    }
```

Comment:
you can consider extracting the low-level string operations to a new method to improve SLAP